sklearn/datasets/_arff_parser.py

```diff
@@ -392,18 +392,21 @@ def _pandas_arff_parser(
     # `"'some string value'"` with pandas.
     single_quote_pattern = re.compile(r"^'(?P<contents>.*)'$")
 
-    def strip_quotes_regex(s):
-        return s.group("contents")
+    def strip_single_quotes(input_string):
+        match = re.search(single_quote_pattern, input_string)
+        if match is None:
+            return input_string
+
+        return match.group("contents")
 
     categorical_columns = [
         name
         for name, dtype in frame.dtypes.items()
         if pd.api.types.is_categorical_dtype(dtype)
     ]
     for col in categorical_columns:
-        frame[col].cat.categories = frame[col].cat.categories.str.replace(
-            single_quote_pattern, strip_quotes_regex, regex=True
-        )
+        frame[col] = frame[col].cat.rename_categories(strip_single_quotes)
```

Comment:
Depending on how https://github.com/pandas-dev/pandas/issues/37643#issuecomment-1045457638 ends up, we may be able to revert back to `cat.categories = `.